### ISSUE TITLE
Player screen goes to completed state on tapping screen off

### ISSUE THREAD

**Issue by michaelcspeed** [meditohq/medito-app#158]
Turning the screen off before starting a session makes the player screen lose its state

**To Reproduce**
1. Go to a session and open the player
2. Turn the screen off
3. Turn it back on again
4. The player looks like the session has completed

**Expected behavior**
The player looks like it does before the screen was turned off

**Devices**
Pixel 2 XL

**Additional context**
The MediaItem is being set to null somehow. (See `lib/widgets/player/player_widget.dart 117 147 and 404`)

**Comment by harshadmanglani** [comment]
I'm going to look into this and try to figure out what's happening.

**Comment by qtdzz** [comment]
@michaelcspeed  @harshadmanglani I actually looked at this and was able to reproduce in the `audio_service` library example as well.
Here is my speculation. This seems to be the current limitation of the audio_service. The service won't start until we actually explicitly tell [it to play the audio](https://github.com/ryanheise/audio_service/blob/b9c89df1f115c8e304af080afa8c8208c56f258f/android/src/main/java/com/ryanheise/audioservice/AudioService.java#L348). In medito, this is the case we open the player without playing it. So the service is not running yet. Then locking the screen would destroy the service (because the service is not running, system will just clean the stopped service, I suppose). This will lead to the following sequence of actions: [AudioService#onDestroy()](https://github.com/ryanheise/audio_service/blob/master/android/src/main/java/com/ryanheise/audioservice/AudioService.java#L552) -> [AudioServicePlugin#onDestroy()](https://github.com/ryanheise/audio_service/blob/master/android/src/main/java/com/ryanheise/audioservice/AudioServicePlugin.java#L697) -> [trigger Dart side client handler `onStopped`](https://github.com/ryanheise/audio_service/blob/b9c89df1f115c8e304af080afa8c8208c56f258f/android/src/main/java/com/ryanheise/audioservice/AudioServicePlugin.java#L900) -> [reset all the state including adding `null` to the MediaItem stream](https://github.com/ryanheise/audio_service/blob/master/lib/audio_service.dart#L662)

I don't see an easy fix here other than a workaround to autoplay the media when users open the player_widget. But it would be better after we implementing the background sound preference feature. https://github.com/meditohq/medito-app/issues/160

For a long term fix, I suggest to hold on until the new version of `audio_service:  0.18`. The new version will make the implementation in Medito much simpler and also fix the issue without auto playing I think. I tested the example of the new version. Though we need a bit of refactoring to support the new version....

**Comment by michaelcspeed** [comment]
From my understanding, the service starts when you call `AudioService.start` which we're doing [here](https://github.com/meditohq/medito-app/blob/develop/lib/widgets/player/player_widget.dart#L525). Otherwise you wouldn't even get the MediaItem in the first place, I assume.

`v0.18` sounds very promising though! It would be great to refactor the audio_player.dart file at some point because it's a bit of a mess.

**Comment by qtdzz** [comment]
> the service starts when you call AudioService.start which we're doing here

Yeah, I realized that. But if you look a bit deeper, it only "starts" on the Dart interface. Under the hood, it calls [`AudioService#init`](https://github.com/ryanheise/audio_service/blob/b9c89df1f115c8e304af080afa8c8208c56f258f/android/src/main/java/com/ryanheise/audioservice/AudioService.java#L81) in the Android side which initializes the service. The `AudioService` in the Android code doesn't start until you call `setState` of the service to "playing". This happens in [audio_player_service#_broadcastState](https://github.com/meditohq/medito-app/blob/develop/lib/audioplayer/audio_player_service.dart#L219) of Medito code. Then, in the Android side of the `audio_service` plugin will [`enterPlayingState`](https://github.com/ryanheise/audio_service/blob/master/android/src/main/java/com/ryanheise/audioservice/AudioService.java#L243-L247) where the Android `AudioService` [actually starts](https://github.com/ryanheise/audio_service/blob/b9c89df1f115c8e304af080afa8c8208c56f258f/android/src/main/java/com/ryanheise/audioservice/AudioService.java#L349). This will prevent it being destroyed when locking screen.


What I am trying to say is that from Medito side, there is nothing wrong, it's just the way how the library currently works. And I suspect this will be fixed (and simplified alot) in the 0.18 version

**Comment by michaelcspeed** [comment]
OK, good digging @qtdzz, thanks for that. I'm happy to put this on hold until the new version is released.

**Comment by harshadmanglani** [comment]
@qtdzz awesome work!!

**Comment by scouech** [comment]
hey @qtdzz @michaelcspeed , any news related to that?

**Comment by michaelcspeed** [comment]
@scouech no, we are waiting for the new version of the audio library that should fix the problem

**Comment by michaelcspeed** [comment]
I've upgraded the libs to the new versions, but not yet checked if the problem is fixed

**Comment by michaelcspeed** [comment]
This is fixed in v2.0.38
